pkg/kubelet/util/util_windows_test.go

```diff
@@ -232,11 +233,50 @@ func testUnixDomainSocket(t *testing.T, label string) {
 	assert.True(t, result, "Unexpected result: false from IsUnixDomainSocket: %v for %s", result, label)
 }
 
+// This is required as on Windows it's possible for the socket file backing a Unix domain socket to
+// exist but not be ready for socket communications yet as per
+// https://github.com/kubernetes/kubernetes/issues/104584
+func testPendingUnixDomainSocket(t *testing.T, label string) {
+	// Create a temporary file that will simulate the Unix domain socket file in a
+	// not-yet-ready state. We need this because the Kubelet keeps an eye on file
+	// changes and acts on them, leading to potential race issues as described in
+	// the referenced issue above
+	f, err := ioutil.TempFile("", "test-domain-socket")
+	require.NoErrorf(t, err, "Failed to create file for test purposes: %v while setting up: %s", err, label)
+	testFile := f.Name()
+	f.Close()
+
+	// Start the check at this point
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		result, err := IsUnixDomainSocket(testFile)
+		assert.Nil(t, err, "Unexpected error: %v from IsUnixDomainSocket for %s", err, label)
+		assert.True(t, result, "Unexpected result: false from IsUnixDomainSocket: %v for %s", result, label)
+		wg.Done()
+	}()
+
+	// Wait a sufficient amount of time to make sure the retry logic kicks in
+	time.Sleep(delayBetweenSuccessiveSocketDials)
```

Comment:
this is asynchronous, have you run this with  `go test -race -c $pkg && stress $pkg.test` and ensure that it’s not flaky? 

([guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md#deflaking-unit-tests))

Reply:
Yes. Here are the commands run and the output (trimmed for visibility):
```
C:\Users\mihai.albert\GolandProjects\kubernetes\pkg\kubelet\util>go test . -c -mod=mod -v -run TestIsUnixDomainSocket -race

C:\Users\mihai.albert\GolandProjects\kubernetes\pkg\kubelet\util>stress ./util.test.exe -test.run TestIsUnixDomainSocket
5s: 0 runs so far, 0 failures
10s: 0 runs so far, 0 failures
15s: 0 runs so far, 0 failures
20s: 8 runs so far, 0 failures
..............
24m10s: 760 runs so far, 0 failures
24m15s: 760 runs so far, 0 failures
24m20s: 762 runs so far, 0 failures
24m25s: 768 runs so far, 0 failures
```